### PR TITLE
Support URL parameters for filters

### PR DESCRIPTION
**Pull request by mattwire**
- Add support for passing URL parameters for filters.
- Update documentation (readme) - I don't understand some of it - @agileware-justin are you able to comment and I'll update so it makes sense to me?

### CODE REVIEW

**Comment by agileware-justin** [comment]
Thanks @mattwire I actually thought that URL params was already available here.

Anyway, continue building out your PR and let us know when it's ready for review.

File: includes/class-cvc-query.php [hunk 1]
```diff
@@ -77,16 +77,19 @@ public function alter_query_parameters( $args ) {
...
 		// hidden filters
 		$fields = $this->cvc->api->call_values( 'DataProcessorFilter', 'get', [
-			'sequential'        => 1,
-			'is_required'       => 1,
 			'is_exposed'        => 1,
 			'data_processor_id' => $id
 		] );
 		foreach ( $fields as $field ) {
-			if ( $this->cvc->dp_options->has_option( $field['name'], Content_Views_CiviCRM_Dp_Option::USER_CONTACT_ID ) ) {
+			if ( $field['is_required'] && $this->cvc->dp_options->has_option( $field['name'], Content_Views_CiviCRM_Dp_Option::USER_CONTACT_ID ) ) {
```
Comment (agileware-pengyi) [inline includes/class-cvc-query.php:89]: Only get the logged in user when the field is required? Sounds good to me.

**Review by agileware-pengyi — COMMENTED** [review]
@mattwire thanks for this PR. I saw a lot of warnings in the error log, so this should help.
The `getfields` call is fixed in other place. Could you please remove/rebase it?
I will leave the readme for @agileware-justin

File: includes/class-cvc-query.php (hunk 1 above)
Comment (mattwire) [reply includes/class-cvc-query.php:89]: :+1:

**Comment by mattwire** [comment]
@agileware-pengyi @agileware-justin This should now be ready for review :-)

**Comment by agileware-justin** [comment]
Readme changes look fine to me, thanks @mattwire

**Comment by agileware-pengyi** [comment]
@mattwire thanks for the update. I've reviewed this PR. Do you want to solve the conflict? Then I will merge this. Thanks

**Comment by agileware-pengyi** [comment]
Never mind. Merged

**Comment by mattwire** [comment]
Thankyou @agileware-pengyi

**Pull request merged**